charms/knative-serving/src/image_management.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+from typing import Dict
+
+import yaml
+
+
+def parse_image_config(image_config: str) -> Dict[str, str]:
+    """Parses config data for a dict of images, returning the parsed value as a dict."""
+    try:
+        images_from_config = yaml.safe_load(image_config)
+    except yaml.YAMLError as err:
+        raise ValueError(
+            f"Encountered error while parsing image config with value '{image_config}'"
```

Comment:
Since this message is passed all the way to the user through the blocked status message, should we add instructions on how to move forward to unblock the unit?
If not here, then we should tell users how to unblock the unit in the charm code.

Reply:
agreed!  added to the blocked message.